src/BinSkim.Rules/ELFRules/BA3003.EnableStackProtector.cs

```diff
@@ -42,46 +46,96 @@ public class EnableStackProtector : ELFBinarySkimmerBase
+                IELF elf = context.ELFBinary().ELF;
+                symbolNames =
+                    new HashSet<string>
+                    (
+                        ELFUtility.GetAllSymbols(elf).Select<ISymbolEntry, string>(sym => sym.Name)
+                    );
 
-            foreach (string stack_chk in this.stack_check_symbols)
+                result = symbolNames.Any(symbol => this.stack_check_symbols.Contains(symbol));
+            }
+            else if (context.IsMachO())
             {
-                if (symbolNames.Contains(stack_chk))
+                MachOBinary mainMachO = context.MachOBinary();
+                foreach (SingleMachOBinary singleMachO in mainMachO.MachOs)
                 {
-                    context.Logger.Log(this,
-                        RuleUtilities.BuildResult(ResultKind.Pass, context, null,
-                            nameof(RuleResources.BA3003_Pass),
-                            context.TargetUri.GetFileName()));
-                    return;
+                    if (this.IsApplicableMachO(singleMachO))
+                    {
+                        symbolNames = new HashSet<string>
+                            (
+                                singleMachO.SymbolTables.SelectMany(st => st.Symbols).Select(s => s.Name)
+                            );
+
+                        result = symbolNames.Any(symbol => this.stack_check_symbols.Contains(symbol));
+
+                        // if any macho fails the check
+                        if (!result)
+                        {
+                            break;
+                        }
+                    }
                 }
+
             }
-            // If we haven't found the stack protector, we assume it wasn't used.
-            context.Logger.Log(this,
-                RuleUtilities.BuildResult(FailureLevel.Error, context, null,
-                    nameof(RuleResources.BA3003_Error),
-                    context.TargetUri.GetFileName()));
+
+            if (result)
+            {
+                context.Logger.Log(this,
+                    RuleUtilities.BuildResult(ResultKind.Pass, context, null,
+                        nameof(RuleResources.BA3003_Pass),
+                        context.TargetUri.GetFileName()));
+            }
+            else
+            {
+                // If we haven't found the stack protector, we assume it wasn't used.
+                context.Logger.Log(this,
+                    RuleUtilities.BuildResult(FailureLevel.Error, context, null,
+                        nameof(RuleResources.BA3003_Error),
+                        context.TargetUri.GetFileName()));
+            }
+        }
+
+        private bool IsApplicableMachO(SingleMachOBinary macho)
+        {
+            return (macho.MachO.FileType == ELFSharp.MachO.FileType.DynamicLibrary
+                || macho.MachO.FileType == ELFSharp.MachO.FileType.Executable
+                || macho.MachO.FileType == ELFSharp.MachO.FileType.Object);

```

Comment:
rewrite to this:
```csharp
return (macho.MachO.FileType == ELFSharp.MachO.FileType.Object ||
    macho.MachO.FileType == ELFSharp.MachO.FileType.Executable ||
    macho.MachO.FileType == ELFSharp.MachO.FileType.DynamicLibrary)
```

let's always order by line length, adding the operator at the end